src/web/services/student.service.ts

```diff
@@ -36,14 +36,25 @@ export class StudentService {
   }
 
   /**
-   * Get a student using their course ID and email by calling API.
+   * Gets student of a course by calling API.
+   *
+   * <p> If both studentEmail and regKey are not provided, get the student of current logged-in user.
+   *
+   * @param courseId courseId of the course
+   * @param studentEmail if provided, get the student of the course of the given email
+   * @param regKey if provided, get the student of the course with regKey
    */
-  getStudentFromCourse(studentEmail: string, courseId: string): Observable<Student> {
-    const paramMap: { [key: string]: string } = {
+  getStudent(courseId: string, studentEmail?: string, regKey?: string): Observable<Student> {
```

Comment:
Do we want to standardize the params between here (which uses courseID, studentEmail) and student profile service's getStudentProfile (which uses studentEmail, courseID)?

Reply:
After all API call are migrated to use service?